### PR TITLE
Allow `trait A: ~const B`

### PR DESCRIPTION
**Pull request by fee1-dead**
What's included: a minimal working change set for `~const` supertraits to work.

r? @oli-obk

### CODE REVIEW

**Comment by fee1-dead** [comment]
@oli-obk is this okay to merge now?

File: compiler/rustc_trait_selection/src/traits/wf.rs [hunk 1]
```diff
@@ -689,6 +696,7 @@ impl<'tcx> WfPredicates<'tcx> {
         &mut self,
         def_id: DefId,
         substs: SubstsRef<'tcx>,
+        remap_constness: bool,
```
Comment (oli-obk) [inline compiler/rustc_trait_selection/src/traits/wf.rs:699]: Rename this function, make it private and create two wrappers around it. One with the original name, passing false, and one with a trailing with_constness `` or similar

**Pull request merged**